counsel.el

```diff
@@ -132,113 +132,100 @@ descriptions.")
 (defvar counsel-async-ignore-re nil
   "Regexp matching candidates to ignore in `counsel--async-filter'.")
 
-(defun counsel--async-command (cmd &optional process-sentinel process-filter)
+(defun counsel-delete-process (&optional name)
```

Comment:
Please move this function back to the original place, so that it's easy to see what changed in the diff. Moving functions around is fine, but only if their content doesn't change.

Reply:
No problem. Should I also move the function to before its first call site in a separate commit, or just leave it put?

Reply:
As you prefer. There won't be a compiler warning if you leave it put. But it might also be nice to group the API part of counsel (to which `counsel-delete-process` belongs) separately from the adaptations part (e.g. `counsel-ag`). Looking at it now, this is the case already, so it's fine not to move it.

Reply:
OK, I'll see if I find anything which can be reordered after I refactor the other points.